pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -430,3 +438,44 @@ def _detect_replacable_format_call(self, node: nodes.Const) -> None:
                 line=node.lineno,
                 col_offset=node.col_offset,
             )
+
+    @utils.only_required_for_messages("consider-refactoring-into-while-condition")
+    def visit_while(self, node: nodes.While) -> None:
+        self._check_breaking_after_while_true(node)
+
+    def _check_breaking_after_while_true(self, node: nodes.While) -> None:
+        """Check that any loop with an else clause has a break statement."""
+        if not isinstance(node.test, nodes.Const):
+            return
+        if not isinstance(node.body[0], nodes.If):
+            return
+        if not isinstance(node.body[0].body[0], nodes.Break):
+            return
+        # Construct error/recommendation message
+        test_node = node.body[0].test
+        if isinstance(test_node, nodes.UnaryOp):
+            msg = test_node.operand.as_string()
+        elif isinstance(test_node, nodes.BoolOp):
+            msg = f"not ({test_node.as_string()})"
+        elif isinstance(test_node, nodes.Name):
+            msg = f"not {test_node.as_string()}"
+        else:
+            lhs = test_node.left
+            ops, rhs = test_node.ops[0]
+
+            msg = f"not {node.body[0].test.as_string()}"
+            if ops == "is":
+                msg = f"{lhs.as_string()} is not {rhs.as_string()}"
+            elif ops == "is not":
+                msg = f"{lhs.as_string()} is {rhs.as_string()}"
+            elif ops == "==":
+                msg = f"{lhs.as_string()} != {rhs.as_string()}"
+            elif ops == "!=":
+                msg = f"{lhs.as_string()} == {rhs.as_string()}"
+
+        self.add_message(
+            "consider-refactoring-into-while-condition",
+            node=node,
+            line=node.lineno,
+            args=(msg,),
```

Comment:
```suggestion
            args=(msg,),
            confidence=HIGH,
```

Reply:
just wondering, what should I specify for confidence moving forward in future contributions?